browser/main/lib/dataApi/attachmentManagement.js

```diff
@@ -227,7 +227,7 @@ function migrateAttachments (markdownContent, storagePath, noteKey) {
  * @returns {String} postprocessed HTML in which all :storage references are mapped to the actual paths.
  */
 function fixLocalURLS (renderedHTML, storagePath) {
-  return renderedHTML.replace(new RegExp('/?' + STORAGE_FOLDER_PLACEHOLDER + '.*?"', 'g'), function (match) {
+  return renderedHTML.replace(new RegExp('/?' + STORAGE_FOLDER_PLACEHOLDER + '(?:\\\/|%5C)[\\w.]+', 'g'), function (match) {
```

Comment:
I've undone the change because I don't remember why I've changed it. And it works with the original one...

Reply:
Oops, now, I understand why :grin:

> ```
> ![#5.jpg](:storage/3b6f8bd6-4edd-4b15-96e0-eadc4475b564/fad71156.jpg)
> ```
> generates
> ```
> <p data-line="1"><img src=":storage/3b6f8bd6-4edd-4b15-96e0-eadc4475b564/fad71156.jpg" alt="#5.jpg" /></p>
> ```

> ~~~
> ```gallery(30h)
> ![#5.jpg](:storage/3b6f8bd6-4edd-4b15-96e0-eadc4475b564/fad71156.jpg)
> ```
> ~~~
> generates
> ```
> <pre class="fence" data-line="1">
>           <span class="filename"></span>
>           <div class="gallery" data-autoplay="undefined" data-height="20">:storage/3b6f8bd6-4edd-4b15-96e0-eadc4475b564/fad71156.jpg</div>
> </pre>
> ```

`new RegExp('/?' + STORAGE_FOLDER_PLACEHOLDER + '.*?"', 'g')` will only match the first image due to the need of the closing `"`.
`new RegExp('/?' + STORAGE_FOLDER_PLACEHOLDER + '(?:\\\/|%5C)[\\w.]+', 'g')` will match both images. The `\\\/` is for posix systems and `%5C` is for windows.
It might need further testing on windows.

Reply:
Couldn't you use the `PATH_SEPARATORS ` variable which is defined at the top of the file?

Reply:
`PATH_SEPARATORS `'s value is `/\\` so it can't be used.

Reply:
could you, at least, use `path.win32.sep` and `path.posix.sep`? That would make clearer what the RegExp means..

Reply:
`path.win32.sep` is `\\` which is far from the needed `%5C`

Reply:
Then i am wondering whether you really need your `%5C` i mean: it dosen't appear everywhere else in the attachment-management, does it?

Reply:
It's the URI encoding of `\`. After I don't know where it's been done.